### ISSUE TITLE
Rename does not update usages in complex patterns.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#9079]
```c#
    public class Expression { }
    public class Constant : Expression
    {
        public readonly int Value;
        public Constant(int Value)
        {
            this.Value = Value;
        }
    }
    public class Plus : Expression
    {
        public readonly Expression Left, Right;     // rename Left here.
        public Plus(Expression Left, Expression Right)
        {
            this.Left = Left;
            this.Right = Right;
        }
    }
    public class X
    {
        public static void Main()
        {
            Expression expr = new Plus(new Plus(new Constant(1), new Plus(new Constant(2), new Constant(3))), new Constant(6));

            if (expr is Plus { Left is Plus p,Right is Constant c })
            {
            }
```

Renaming Plus.Left does not update reference in the complex pattern.

**Comment by jaredpar** [comment]
Reopen pending unit tests.

**Comment by gafter** [comment]
Fixed in features/patterns via #9440.
